packages/extractor-vue/src/vue-extractor.ts

```diff
@@ -0,0 +1,28 @@
+import * as vueCompiler from "@vue/compiler-sfc"
+// @ts-expect-error issue with typings, will be fixed shortly
+import babel from "@lingui/cli/api/extractors/babel"
+import type { ExtractorCtx, ExtractorType } from "@lingui/conf"
+
+export const vueExtractor: ExtractorType = {
+  match(filename: string) {
+    return filename.endsWith(".vue")
+  },
+  extract(
+    filename: string,
+    code: string,
+    onMessageExtracted,
+    ctx: ExtractorCtx
+  ) {
+    const {
+      descriptor: { script, scriptSetup },
+    } = vueCompiler.parse(code, { sourceMap: true, filename })
+
+    const finalCode = `${script?.content} ${scriptSetup?.content}`
+    const sourcemaps = `${script?.map} ${scriptSetup?.map}`
```

Comment:
im not sure this is going to work like that

You actually can run `babel.extract` few times on each virtual file extracted from `.vue` if you don't want to "play" with sourcemaps.

Reply:
> im not sure this is going to work like that
> 
> You actually can run `babel.extract` few times on each virtual file extracted from `.vue` if you don't want to "play" with sourcemaps.

@thekip Yeah will probably have to do something like that to make sourcemaps work